src/boot/efi/boot.c

```diff
@@ -1651,6 +1651,51 @@ static EFI_STATUS image_start(EFI_HANDLE parent_image, const Config *config, con
                 loaded_image->LoadOptionsSize = (StrLen(loaded_image->LoadOptions)+1) * sizeof(CHAR16);
 
 #ifdef SD_BOOT_LOG_TPM
+                UINTN i, j;
+                UINTN length;
+                CHAR16* begin, *end;
+
+                EFI_FILE* root_dir = LibOpenRoot(entry->device);
```

Comment:
also, please don't mix function invocations and variable declarations in the same line, also see CODING_STYLE.